### PR TITLE
Restructure bazelrc for reproducible builds

### PR DESCRIPTION
**Pull request by GMNGeoffrey**
Moves the bulk of our bazelrc to a separate directory and then imports
it in the workspace bazelrc. This allows us to remove the workspace rc
from CI build scripts, which means that when running them locally we're
actually running with the same flags for reproducible builds and cache
hits.

### CODE REVIEW

**Comment by GMNGeoffrey** [comment]
Ping @stellaraccident?

File: .bazelrc [hunk 1]
```diff
@@ -12,276 +12,10 @@
...
-# Disables TensorFlow eigen bloat and reduces compile times.
-build:windows --define=override_eigen_strong_inline=true
-
-# Another TensorFlow flag from their config script.
-build:windows --define with_default_optimizations=true
-
-###############################################################################
+# Import the main bazelrc config. This is in a separate file so that it's easy
+# to turn off all user bazelrc options by specifying
+# `--nosystem_rc --nohome_rc --noworkspace_rc --bazelrc=build_tools/bazel/iree.bazelrc`
```
Comment (ScottTodd) [inline .bazelrc:17]: "easy to turn off"
> 4 magic flags

Oh Bazel..

File: .bazelrc [hunk 2]
```diff
@@ -12,276 +12,10 @@
...
+try-import %workspace%/build_tools/bazel/iree.bazelrc
```
Comment (ScottTodd) [inline .bazelrc:18]: I'm on the fence about this being nested under `./build_tools/bazel/`.  Each of the other `*.bazelrc` files is in the root... but we'd like to minimize the number of files in the root if possible. I guess the parallel to https://github.com/google/iree/blob/main/build_tools/cmake/iree_copts.cmake makes this less of an issue?

Comment (GMNGeoffrey) [reply .bazelrc:18]: The other bazelrc are added by users though and in the `.gitignore`. They seem much more akin to user dot files and as untracked files need to be easy to find. My rationale here was that we should stick as much build_tools related stuff under build_tools. Honestly I thought you'd be thrilled that I didn't add another file at root :stuck_out_tongue:

File: .bazelrc (hunk 1 above)
Comment (GMNGeoffrey) [reply .bazelrc:17]: Good point. I meant "possible" :-D

**Pull request merged**